oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/ElasticsearchPropertyIndexTest.java

```diff
@@ -118,12 +117,15 @@ public void indexSelection() throws Exception {
         root.commit();
 
         String propaQuery = "select [jcr:path] from [nt:base] where [propa] = 'foo'";
-        assertThat(explain(propaQuery), containsString("elasticsearch:test1"));
-        assertThat(explain("select [jcr:path] from [nt:base] where [propc] = 'foo'"), containsString("elasticsearch:test2"));
 
-        assertQuery(propaQuery, Arrays.asList("/test/a", "/test/b"));
-        assertQuery("select [jcr:path] from [nt:base] where [propa] = 'foo2'", singletonList("/test/c"));
-        assertQuery("select [jcr:path] from [nt:base] where [propc] = 'foo'", singletonList("/test/d"));
+        assertEventually(() -> {
+            assertThat(explain(propaQuery), containsString("elasticsearch:test1"));
+            assertThat(explain("select [jcr:path] from [nt:base] where [propc] = 'foo'"), containsString("elasticsearch:test2"));
+
+            assertQuery(propaQuery, Arrays.asList("/test/a", "/test/b"));
+            assertQuery("select [jcr:path] from [nt:base] where [propa] = 'foo2'", singletonList("/test/c"));
+            assertQuery("select [jcr:path] from [nt:base] where [propc] = 'foo'", singletonList("/test/d"));
+        }, 10000);
```

Comment:
Should we define a constant for `10000`? It seems to be used at multiple places. If this is related to BULK_FLUSH_INTERVAL_MS_DEFAULT, may be we could use an expression to calculate it like -
```
private static final int TIMEOUT  = BULK_FLUSH_INTERVAL_MS_DEFAULT * 3;
```